ubuntu_image/builder.py

```diff
@@ -333,6 +340,9 @@ def populate_filesystems(self):
                     os.path.join(part_dir, filename)
                     for filename in os.listdir(part_dir)
                     )
+                # XXX Make sure the $PATH gets propagated to the mcopy(1)
+                # subprocess.  For unknown reasons it might not yet be set
+                # up.  See PR#46 for details.
```

Comment:
Hm, unless I miss something the reason is the following:
```
$ python3 
Python 3.5.2 (default, Jul  5 2016, 12:43:10) 
[GCC 5.4.0 20160609] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from ubuntu_image.helpers import run
>>> run("env", env=dict(MTOOLS_SKIP_CHECK='1'), stdout=None)
MTOOLS_SKIP_CHECK=1
CompletedProcess(args=['env'], returncode=0, stderr='')
```
I.e. once you set `env` to something it replaces the existing env and does not merge it(compare the output to a run() without env, here all the user env is still there. This means of course that the custom PATH that is set by the snapcraft wrapper will not work. Also the LD_LIBRARY_PATH etc are gone. Fwiw, without inheriting env (e.g. via `run("env", env=dict(MTOOLS_SKIP_CHECK='1').update(os.environ), stdout=None)`) here and on the other places that set `env` the snap won't work.